iodata/orbitals.py

```diff
@@ -168,32 +202,115 @@ def spinpol(self) -> float:
         if self.kind == "restricted":
-            nbeta = np.clip(self.occs, 0, 1).sum()
-            return abs(self.nelec - 2 * nbeta)
+            if self.occs_aminusb is None:
+                # heuristics ...
+                if (self.occs == self.occs.astype(int)).all():
+                    # restricted open-shell HF/KS
+                    nbeta = np.clip(self.occs, 0, 1).sum()
+                    return abs(self.nelec - 2 * nbeta)
+                # restricted closed-shell natural orbitals
+                return 0.0
+            return self.occs_aminusb.sum()
         return abs(self.occsa.sum() - self.occsb.sum())
 
     @property
     def occsa(self):
-        """Return alpha occupation numbers."""
+        """Return alpha occupation numbers.
+
+        Notes
+        -----
+        For restricted wavefunctions, in-place assignment to occsa will not
+        work. In this case, the array is derived from ``mo.occs`` and optionally
+        ``mo.occs_aminub``. To avoid that in-place assignment of occsa is
+        silently ignored, it is returned as a non-writeable array. To change
+        occsa, one can assign a whole new array, e.g. ``mo.occsa = new_occsa``
+        will work, while ``mo.occsa[1] = 0.3`` will not.
+
+        """
         if self.kind == "generalized":
             raise NotImplementedError
         if self.occs is None:
             return None
         if self.kind == "restricted":
-            return np.clip(self.occs, 0, 1)
+            if self.occs_aminusb is None:
+                # heuristics ...
+                if (self.occs == self.occs.astype(int)).all():
+                    # restricted open-shell HF/KS
+                    result = np.clip(self.occs, 0, 1)
+                else:
+                    # restricted closed-shell natural orbitals
+                    result = self.occs / 2
+            else:
+                result = (self.occs + self.occs_aminusb) / 2
+            result.flags.writeable = False
+            return result
         return self.occs[: self.norba]
 
+    @occsa.setter
+    def occsa(self, occsa):
+        if self.kind == "generalized":
+            raise NotImplementedError
+        if self.kind == "restricted":
+            occsa = np.array(occsa)
+            if self.occs is None:
+                self.occs = occsa
+                self.occs_aminusb = occsa.copy()
+            else:
+                occsb = np.array(self.occsb)
+                self.occs = occsa + occsb
+                self.occs_aminusb = occsa - occsb
+        else:
+            self.occs[: self.norba] = occsa
+
     @property
     def occsb(self):
-        """Return beta occupation numbers."""
+        """Return beta occupation numbers.
+
+        Notes
+        -----
+        For restricted wavefunctions, in-place assignment to occsb will not
+        work. In this case, the array is derived from ``mo.occs`` and optionally
+        ``mo.occs_aminub``. To avoid that in-place assignment of occsb is
```

Comment:
mo.occs_aminusb (typo)

Reply:
Thanks! Both occurrences of this typo were fixed.